python/tank/bootstrap/cached_configuration.py

```diff
@@ -282,6 +289,17 @@ def update_configuration(self):
                     self.path.as_shotgun_dict()
                 )
 
+    def cleanup_backup_folders(self):
+        try:
+            if self.config_backup_folder_path is not None:
+                filesystem.delete_folder(self.config_backup_folder_path)
+                self.config_backup_folder_path = None
+            if self.core_backup_folder_path is not None:
```

Comment:
Could you write this as two try/catch blocks, this way failing to delete the config doesn't mean automatically that we can't delete the core.